### ISSUE TITLE
About imagenet-C

### ISSUE THREAD

**Issue by z1358** [Lily-Le/EcoTTA#1]
Thanks for your implementation!
For the cifar10-C dataset with backbone wideresnet28, i get the avg. err at 18.22. And the batch size for CTTA is 200, with the lr of 5e-3.
However, for the imagenet-C dataset, no matter how the learning rate or batch size is set, i get the avg. err above 90.0 almost every time.
The warm-up process is the same as in the code main_imagenet.py. The main log is :
.....
Evaluating Network.....
Test set: Epoch: 0, Average loss: 0.0044, Accuracy: 0.7237, Time consumed:203.34s
.....
Evaluating Network.....
Test set: Epoch: 1, Average loss: 0.0056, Accuracy: 0.6636, Time consumed:180.85s
.....
Evaluating Network.....
Test set: Epoch: 2, Average loss: 0.0051, Accuracy: 0.6919, Time consumed:181.79s
.....
Evaluating Network.....
Test set: Epoch: 3, Average loss: 0.0051, Accuracy: 0.6914, Time consumed:165.96s
saving weights file to ./checkpoint_pretrain/resnet50/warmup_bs256_lr0.05/resnet50-3-regular.pth

I would like to know if you are able to reproduce the error rates in the paper.
Looking forward to your reply.

**Comment by Lily-Le** [comment]
@z1358  Actually me too.  I tried different tta lr / lambda for imagnet-c, but the results were unsatisfactory.  I am doing more experiments gradually (when I can spare time for it) to see the possible reasons.  I'm not sure whether it's because of the data transformation for imagenet-c or other problems that leads to performance degradation.  
In my case,  the warmup stage for imagenet is:
<img src="https://github.com/Lily-Le/EcoTTA/assets/21274476/030c42fb-5a2f-4f1e-8494-4003b983b19d" width="500"/>
The test ACC(performance on the original imagenet dataset without augmentation) stays almost the same for the 3 warmup epochs. But the tta results are bad. 

And, yes, my current result for wrs28-10 cifar10 is about 18.48%. The results fluctuate for different warmup checkpoints, but for now, it still lags behind reported in the paper. That's why I only updated the WRS40-CIFAR100 result on Readme for now.    

I think that there may be other points that may influence the final results. Maybe somewhere I didn't loyally reproduce the paper, or there are other things to be watched out like the different training tricks.  From my experience, the performance of the warmup stage greatly influences the tta performance.   
If you find out anything that may lead to the performance mismatch, pls let me know. It's not an official implementation and may have some problems.

**Comment by z1358** [comment]
@Lily-Le 
Okay, I get it. I'll try more experiments too. If I find anything new, I'll share it with you.
Thanks again for your positive implementation.

**Comment by daeunni** [comment]
@z1358 Hi, did you get the similar avg result from CIFAR10?
